src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -223,6 +223,36 @@ public static unsafe char AnsiCharToWideChar(byte nativeValue)
             return PInvokeMarshal.AnsiCharToWideChar(nativeValue);
         }
 
+        internal static unsafe char* StringToBstrBuffer(string str)
+        {
+            if (str == null)
+                return null;
+
+            int stringLength = str.Length;
+
+            char* buffer = (char*)Marshal.AllocCoTaskMem(sizeof(char) * (stringLength + 1) + 4);
+
+            fixed (char* pStr = str)
+            {
+                int size = stringLength * sizeof(char);
+                char* firstCharPtr = buffer + 4;
+                Buffer.MemoryCopy(pStr, firstCharPtr, size, size);
+                *(buffer + stringLength + 2) = '\0';
+                int* pLength = (int*)buffer;
+                *pLength = stringLength * 2;
+            }
+            return buffer;
+        }
+
+        internal static unsafe string BstrBufferToString(char* buffer)
+        {
+            if (buffer == null)
+                return null;
+
+            int stringLength = *(int*)buffer;
```

Comment:
This does not look right. The BSTR pointer points at the first char*, the length is at negative offset.

Reply:
Ok. That's my overlook. I try to find that online, but cannot find any explicit reference. Will change accordingly.